wallet-new/src/Cardano/Wallet/API/V1/Handlers/Accounts.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators     #-}
+
+module Cardano.Wallet.API.V1.Handlers.Accounts where
+
+import           Universum
+
+import           Servant
+
+import           Cardano.Wallet.WalletLayer.Types (PassiveWalletLayer (..))
+import qualified Cardano.Wallet.WalletLayer.Types as WalletLayer
+
+import           Cardano.Wallet.API.Request
+import           Cardano.Wallet.API.Response
+import qualified Cardano.Wallet.API.V1.Accounts as Accounts
+import           Cardano.Wallet.API.V1.Types
+import qualified Cardano.Wallet.Kernel.DB.Util.IxSet as KernelIxSet
+import qualified Data.IxSet.Typed as IxSet
+
+handlers :: PassiveWalletLayer IO -> ServerT Accounts.API Handler
+handlers w =  deleteAccount w
+         :<|> getAccount w
+         :<|> listAccounts w
+         :<|> newAccount w
+         :<|> updateAccount w
+         :<|> redeemAda w
+
+deleteAccount :: PassiveWalletLayer IO
+              -> WalletId
+              -> AccountIndex
+              -> Handler NoContent
+deleteAccount layer wId accIdx = do
+    res <- liftIO $ WalletLayer.deleteAccount layer wId accIdx
+    case res of
+         Left e   -> throwM e
```

Comment:
`throwM` is going to throw the exception in `IO`, which will be reported as a `500` error unless caught/handled by the upstream exception handling mechanism. Is that what we want?

Reply:
@parsonsmatt Not at all 😉 In fact, this code has to be intended as provisional. In the time frame our squad got to bring all the functionalities in place for the new data layer, there wouldn't be enough time to properly deal with exceptions. So what I did was to basically trying to have a sane error handling down the guts of the WalletLayer and Kernel (or semi-sane, depending from the point of views 😁 ) and leave this bits unhandled for your squad to potentially help us with the integration work. A similar concern was also raised by @edsko , and whilst I gave him a similar explanation I have failed to actually morph that into an actionable, i.e. a YouTrack ticket 😀 

Will do now -- thanks for keeping me honest!

Reply:
No, I'm glad we're thinking about this stuff :smile: This stuff definitely falls into the "not checked by GHC lmao" thing, so we have to be extra careful

Reply:
> "not checked by GHC lmao"

Or, alternatively, my new favourite `SORRY IT BROKE LOL` :trollface: